src/packaging/_manylinux.py

```diff
@@ -232,9 +237,9 @@ def platform_tags(linux: str, arch: str) -> Iterator[str]:
             glibc_version = _GLibCVersion(glibc_max.major, glibc_minor)
             tag = "manylinux_{}_{}".format(*glibc_version)
             if _is_compatible(tag, arch, glibc_version):
-                yield linux.replace("linux", tag)
+                yield f"{tag}_{arch}"
```

Comment:
> What's the specific reason for the change?

I think it's cleaner, same as musllinux.

> I'm a little concerned someone is relying on the old semantics.

I don't think this is an issue here as it's a function from a private module.

I will none the less add some check (that was missing) to ensure that `sysconfig.get_platform()`starts with `linux_` before calling `_manylinux.platform_tags()`.
In the case of musllinux, it's clearly marked as a pre-requisite but was not enforced.
In the case of manylinux, it would just have done something unexpected.